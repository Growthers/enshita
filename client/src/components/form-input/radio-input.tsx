import { forwardRef } from "react";
import errorStyles from "~/styles/components/form-input/input.module.scss"
import styles from "~/styles/components/form-input/radio.module.scss"
import type { RadioInputProperties } from "./type/model";

// eslint-disable-next-line react/display-name
const RadioInput = forwardRef<HTMLInputElement, RadioInputProperties>(({ inputStyles, id, value, name, current, total, error, ...rest }, ref) => (
  <>
    <input ref={ref} id={id} name={id} className={inputStyles} type="radio" value={value} {...rest} />
    <div className={styles["enshita-radio-input-label-box"]}>
      <label htmlFor={id}>{name}</label>
      <p className={styles["enshita-radio-input-label-text"]}>{current}</p>
      <p>{total}</p>
    </div>
    {error && (
      <p aria-label="error message" className={errorStyles["enshita-input-error"]}>
        {error.message}
      </p>
    )}
  </>
))

export { RadioInput }
