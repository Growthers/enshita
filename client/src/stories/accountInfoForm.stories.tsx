// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
