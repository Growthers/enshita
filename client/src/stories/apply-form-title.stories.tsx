import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
