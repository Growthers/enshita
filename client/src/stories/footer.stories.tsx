// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
