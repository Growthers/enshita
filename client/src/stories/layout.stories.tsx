import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
