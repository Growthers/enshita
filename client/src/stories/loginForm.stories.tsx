// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
