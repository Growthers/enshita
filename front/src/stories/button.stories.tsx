import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
