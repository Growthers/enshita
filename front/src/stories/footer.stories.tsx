import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
