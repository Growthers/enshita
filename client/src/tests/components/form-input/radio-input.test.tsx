import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/form-input/radio-input.stories";

const { Default, Error } = composeStories(stories);

describe("(components) radio button input", () => {
  test("label should be test", () => {
    const { getByLabelText } = render(<Default />);
    expect(getByLabelText("test")).toBeInTheDocument();
  });
  test("error message is exist", () => {
    const { getByLabelText } = render(<Error />);
    expect(getByLabelText("error message")).toBeInTheDocument();
  });
  test("(normal) take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("(error) take snapshot", () => {
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
  });
});
