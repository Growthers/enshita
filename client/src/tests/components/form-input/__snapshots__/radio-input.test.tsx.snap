// Vitest Snapshot v1

exports[`(components) radio button input > (error) take snapshot 1`] = `
<div>
  <input
    aria-label="test"
    id="test"
    name="test"
    type="radio"
    value="test"
  />
  <div
    class="_enshita-radio-input-label-box_cv769_1"
  >
    <label
      for="test"
    >
      Test
    </label>
    <p
      class="_enshita-radio-input-label-text_cv769_12"
    >
      2
    </p>
    <p>
      5
    </p>
  </div>
  <p
    aria-label="error message"
    class="_enshita-input-error_k8dqh_1"
  >
    選択する必要があります
  </p>
</div>
`;

exports[`(components) radio button input > (normal) take snapshot 1`] = `
<div>
  <input
    aria-label="test"
    id="test"
    name="test"
    type="radio"
    value="test"
  />
  <div
    class="_enshita-radio-input-label-box_cv769_1"
  >
    <label
      for="test"
    >
      Test
    </label>
    <p
      class="_enshita-radio-input-label-text_cv769_12"
    >
      2
    </p>
    <p>
      5
    </p>
  </div>
</div>
`;
